### ISSUE TITLE
Reads the entire word when it is modified, 

### ISSUE THREAD

**Issue by javidominguez** [nvaccess/nvda#17326]
### Is your feature request related to a problem? Please describe.
With SpeakTypedWords enabled, when I go back to correct a typing error, NVDA speaks only the last chunk typed. 
Example: 
1. Type a Word as _featrure_ (Note that it is written wrong) and press space. NVDA Will speak the Word. 
2. Backtrack by deleting characters until first R. Now the word will be _feat_. 
3. Complete the word _feature_ correctly and press space. NVDA will speak _ure_ instead of the full word. 

### Describe the solution you'd like
With SpeakTypedWords enabled, NVDA should always speak the entire word instead of just the last characters typed. 

### Describe alternatives you've considered
This addon provides a solution to the issue. Many users have been using it for months without reporting errors and some have requested that this should be the way NVDA works by default. 
https://github.com/javidominguez/speakTypingWords/
The original code for the addon is from @Paulber19 and @Leonardder. 

### Additional context
I don't know if it is the best way to implement it and I don't feel able to incorporate it myself into the core. Could someone else take care of it? With this code or a better one if possible. I and many users would like this way for NVDA to speak the typed words.

**Comment by CyrilleB79** [comment]
The issue has already been submitted: #8065, so I'll close as duplicate.

Note that there has even been a tentative of PR from @LeonarddeR, #8110, which has not reached the master branch unfortunately. This could be of use for anyone who wants to revive this fix.
